app/controllers/appointments.js

```diff
@@ -0,0 +1,276 @@
+          timeslot: this.getTimeSlotOf(r)
+      });
+    });
+  },
+
+  formatTime(hour, minute) {
+    return moment().set({ hour, minute }).format('HH:mm');
+  },
+
+  getTimeStringOf(record) {
+    const isPreset = record.get('constructor.modelName') === 'appointment-slot-preset';
+    if (isPreset) {
+      return this.formatTime(record.get('hours'), record.get('minutes')); // Preset
+    }
+    return moment.utc(record.get('timestamp')).local().format('HH:mm'); // Special day
+  },
+
+  getDateStringOf(slot) {
+    return moment.utc(slot.get('timestamp')).local().format('dddd Do MMM YYYY');
+  },
+
+  getTimeSlotOf(record) {
+    const time = this.getTimeStringOf(record);
+    return this.get('timeSlots').find(ts => ts.time === time);
+  },
+
+  getOpenPresetTimeslot(dayNumber) {
+    return this.get('timeSlots').find(ts => {
+      return !this.get('appointmentSlotPresets').find(record => record.get('day') === dayNumber &&
+        record.get('hours') === ts.get('hours') &&
+        record.get('minutes') === ts.get('minutes')
+      );
+    });
+  },
+
+  getOpenDateTimeslot(date) {
+    return this.get('timeSlots').find(ts => {
+      return !this.get('appointmentSlots').find(record => {
+        let recordDate = new Date(record.get('timestamp'));
+        return (this.sameDay(date, recordDate) && record.get('quota') > 0 && this.getTimeStringOf(record) === ts.get('time'));
+      });
+    });
+  },
+
+  sameDay(d1, d2) {
+    return d1.getFullYear() === d2.getFullYear() &&
+      d1.getMonth() === d2.getMonth() &&
+      d1.getDate() === d2.getDate();
+  },
+
+  // ---- Controller implementation
+
+  loadData() {
+    this.showLoadingSpinner();
+    RSVP.all([
+      this.store.findAll('appointment_slot_preset'),
+      this.store.findAll('appointment_slot')
+    ])
+    .then(([appointmentSlotPresets, appointmentSlots]) => {
+      this.set('appointmentSlotPresets', appointmentSlotPresets);
+      this.set('appointmentSlots', appointmentSlots);
+    })
+    .catch(() => this.showError("Failed to load appointments", () => history.back()))
+    .finally(() => this.hideLoadingSpinner());
+  },
+
+  actions: {
+
+    back() {
+      this.back();
+    },
+
+    showSpecial() {
+      this.set('selectedTab', 'special');
+    },
+
+    showDefault() {
+      this.set('selectedTab', 'presets');
+    },
+
+    toggleOptionsMenu(obj) {
+      obj.set('showOptionsMenu', !obj.get('showOptionsMenu'));
+    },
+
+    showDateCreationPopup() {
+      this.set('dateCreationPopupVisible', true);
+    },
+
+    cancelDateCreation() {
+      this.set('selectedDate', null);
+      this.set('dateCreationPopupVisible', false);
+      this.set('incompleteForm', false);
+      this.set('dateDescription', '');
+    },
+
+    createNewSpecialDate() {
+      if (!this.get('dateDescription') || !this.get('selectedDate')) {
+        this.set('incompleteForm', true);
+        return false;
+      }
+
+      const note = this.get('dateDescription');
```

Comment:
Would it be better yo use `let` here?? As this value will be changed will be every controller object. If it's really a constant than should we move it to `environment.js` file or may be `constants.js`?

Same for all other constants